pkg/asset/installconfig/aws/validation.go

```diff
@@ -298,8 +299,17 @@ func validateServiceEndpoints(fldPath *field.Path, region string, services []aws
 		}
 		conn.Close()
 	}
-	if _, partitionFound := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region); partitionFound {
-		return allErrs
+
+	if partition, partitionFound := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region); partitionFound {
+		if _, ok := partition.Regions()[region]; ok {
+			return allErrs
+		} else if len(services) == 0 {
+			err2 := validateRegion(region)
+			if err2 != nil {
```

Comment:
No need to use `err2` here instead of `err`.
```suggestion
			if err := validateRegion(region); err != nil {
```